org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationPatching.cs

```diff
@@ -0,0 +1,442 @@
+        AllowMultiple = false,
+        AllowAddingNewValues = false,
+        DefaultValue = Rock.SystemGuid.DefinedValue.TRANSACTION_TYPE_CONTRIBUTION,
+        DefinedTypeGuid = Rock.SystemGuid.DefinedType.FINANCIAL_TRANSACTION_TYPE
+    )]
+    [BooleanField("Include TMBT", defaultValue: false)]
+    [AccountField("Financial Account", "If syncing a total amount given which fund should we sync from")]
+    public class HubspotIntegrationPatching : IJob
+    {
+        private string key { get; set; }
+        private List<HSContactResult> contacts { get; set; }
+        private int request_count { get; set; }
+        private string businessUnit { get; set; }
+
+        /// <summary> 
+        /// Empty constructor for job initialization
+        /// <para>
+        /// Jobs require a public empty constructor so that the
+        /// scheduler can instantiate the class whenever it needs.
+        /// </para>
+        /// </summary>
+        public HubspotIntegrationPatching()
+        {
+        }
+
+        /// <summary>
+        /// Job that will run quick SQL queries on a schedule.
+        /// 
+        /// Called by the <see cref="IScheduler" /> when a
+        /// <see cref="ITrigger" /> fires that is associated with
+        /// the <see cref="IJob" />.
+        /// </summary>
+        public virtual void Execute(IJobExecutionContext context)
+        {
+            JobDataMap dataMap = context.JobDetail.JobDataMap;
+
+            // Get HS API Token
+            string attrKey = dataMap.GetString("AttributeKey");
+            string attrValue = GlobalAttributesCache.Get().GetValue(attrKey);
+            key = Encryption.DecryptString(attrValue);
+
+            var current_id = 0;
+
+            PersonService personService = new PersonService(new RockContext());
+
+            //Get Hubspot Properties in Rock Information Group
+            //This will allow us to add properties temporarily to the sync and then not continue to have them forever
+            var propClient = new RestClient("https://api.hubapi.com/crm/v3/properties/contacts?properties=name,label,createdUserId,groupName,options,fieldType");
+            propClient.Timeout = -1;
+            var propRequest = new RestRequest(Method.GET);
+            propRequest.AddHeader("Authorization", $"Bearer {key}");
+            IRestResponse propResponse = propClient.Execute(propRequest);
+            WriteToLog( "Is propResponse Null? : " + propResponse.IsNull().ToString() );
+            var props = new List<HubspotProperty>();
+            var tmbtProps = new List<HubspotProperty>();
+            var propsQry = JsonConvert.DeserializeObject<HSPropertyQueryResult>(propResponse.Content);
+            props = propsQry.results;
+            WriteToLog( "Is propsQry Null? : " + propsQry.IsNull().ToString() );
+            WriteToLog( "Is propsQry.results Null? : " + propsQry.results.IsNull().ToString() );
+            WriteToLog( "Is props Null? : " + props.IsNull().ToString() );
```

Comment:
The `WriteToLog` calls should probably be commented back out since they are used for debugging. These ones also need to be updated to replace the non-functional `.IsNull()` (which is also marked as obsolete).